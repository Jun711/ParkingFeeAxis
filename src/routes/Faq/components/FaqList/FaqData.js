export default FaqData = [
  {
    key: '1',
    text: 'This app allows you to find nearby meter parking spot by searching for ' +
      'a location or move the center flag to a location.'
  },
  {
    key: '2',
    text: 'The markers / flags on the map indicate a meter parking spot.'
  },
  {
    key: '3',
    text: 'There are three types of parking flags. Green is the cheapest, ' +
    'yellow is the average while red is the most expensive nearby parking meter rate.'
  },
  {
    key: '4',
    text: 'If you pan around the map, you will discover parking meter around a center point.'
  },
  {
    key: '5',
    text: 'This app parking meter data comes from Vancouver Open Data Catalogue.'
  },
  {
    key: '6',
    text: 'This app requires location permission and gps to work.'
  },
  {
    key: '7',
    text: 'If no parking spots show up nearby a specific location, it may mean ' +
      'that there is no meter parkings nearby.'
  },
  {
    key: '8',
    text: 'Your location data is not collected by this app.'
  },
  {
    key: '9',
    text: 'If you find any bug or anomaly, feel free to contact me at ninja@ninja.com.'
  },
  {
    key: '10',
    text: 'If you would like to see more features, feel free to contact me at ninja@ninja.com.'
  },
]
