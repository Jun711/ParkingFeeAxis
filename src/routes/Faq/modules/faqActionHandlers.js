import update from "immutability-helper/index";

const errorMsg = [{'key': 0, 'text': 'Error: Faq Loading timeout'}];

export function handleLoadFaqList(state, action) {
  return update(state, {
    faqLoaded: {
      $set: true
    },
    faqList: {
      $set: action.payload
    }
  })
}

export function handleLoadFaqListError(state) {
  // TODO can make a global error page
  return update(state, {
    faqLoaded: {
      $set: true
    },
    faqList: {
      $set: errorMsg
    }
  })
}