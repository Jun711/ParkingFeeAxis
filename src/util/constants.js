// config
export const LATITUDE_DELTA = 0.005
export const defaultAjaxTimeout = 30000
export const DEFAULT_LATITUDE = 49.2820
export const DEFAULT_LONGITUDE = -123.1171

// location permission constants
export const PERMISSION_REQ_TITLE = 'Parking App Location Permission'
export const PERMISSION_REQ_MSG = 'Parking App needs access to your location to locate nearby parking.'

// toast
export const TOAST_HEADER_Y_OFFSET = 145
export const TOAST_HEADER_X_OFFSET = 0
export const LOCATION_ERROR_MSG_ONE = 'Please enable location service.'
export const LOCATION_ERROR_MSG_TWO = 'Location service is unavailable.'
export const LOCATION_ERROR_DEFAULT_MSG = 'Location request timeout.'

// about
export const APP_NAME = 'Path'
export const APP_VERSION = 'Version 1.0.0'

// time
export const WEEKDAY_9_6 = 'weekdayOffice'
export const WEEKDAY_6_PLUS = 'weekdayAfterOffice'
export const SAT_9_6 = 'satOffice'
export const SAT_6_PLUS = 'satAfterOffice'
export const SUN_9_6 = 'sunOffice'
export const SUN_6_PLUS = 'sunAfterOffice'
export const FREE_PARKING = 'freeParking'
export const FREE_PARKING_LIMIT = 'until 9am'
export const NO_TIME_LIMIT_TEXT = 'no limit'


// min and max rate
export const MIN_PARKING_RATE = 100
export const MAX_PARKING_RATE = -1

// color
export const LOADER_COLOR = '#515c6d'
export const THEME_COLOR = '#38414e'
export const TEXT_COLOR = '#000'
// export const CENTRE_MARKER_COLOR = '#8585ad'
export const CURRENT_MARKER_COLOR = '#98a3b3'
export const CENTRE_MARKER_COLOR ='#515c6d'
export const WHITE_COLOR = '#fff'

// font size
export const TITLE_FONT_SIZE = 20
export const PRIMARY_FONT_SIZE = 18
export const SECONDARY_FONT_SIZE = 14
export const APP_NAME_FONT_SIZE = 15
export const APP_VERSION_FONT_SIZE = 10

// search bar
export const SEARCH_PLACEHOLDER = 'Search'
export const SEARCH_INPUT_KEY = 'headerSearchBar'

// error message
export const CONTACT_INFO_LOADING_ERROR = 'Contact info loading timeout'
export const FAQ_LOADING_ERROR = 'FAQ list loading timeout'
