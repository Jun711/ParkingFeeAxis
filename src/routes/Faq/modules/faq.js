// contains actions and handlers
import update from 'immutability-helper'
import constants from './actionConstants'
import {} from '../../../util/constants'
import { ToastAndroid } from 'react-native'
import timeout from '../../../util/timeout'
import {
  PARKING_FAQS,
} from '../../../util/constants'
import {
  handleLoadFaqList,
  handleLoadFaqListError
} from './faqActionHandlers'

//-------------------------------
//Constants
//-------------------------------
const {
  LOAD_FAQ_LIST,
  LOAD_FAQ_LIST_ERROR
} = constants

//-------------------------------
// Utility function
//-------------------------------


//-------------------------------
// Actions
//-------------------------------
export function loadFaqList() {
  return async (dispatch) => {
    try {
      const response = await timeout(fetch(PARKING_FAQS))
      const responseJson = await response.json()
      dispatch({
        type: LOAD_FAQ_LIST,
        payload: responseJson
      })
    } catch (error) {
      dispatch({
        type: LOAD_FAQ_LIST_ERROR,
        payload: error
      })
    }
  }
}

//-------------------------------
//Action Handlers
//-------------------------------
const ACTION_HANDLERS = {
  LOAD_FAQ_LIST: handleLoadFaqList,
  LOAD_FAQ_LIST_ERROR: handleLoadFaqListError,
}

//-------------------------------
// Initialization
//-------------------------------
export const initialState = {
  faqLoaded: false,
  faqList: []
}

export function FaqReducer(state = initialState, action) {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}