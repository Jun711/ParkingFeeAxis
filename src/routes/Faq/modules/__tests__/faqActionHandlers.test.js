import { handleLoadFaqList } from '../faqActionHandlers'
import { initialState } from '../faq'
import constants from '../actionConstants'
import FaqData from '../../components/FaqList/FaqData'

const {LOAD_FAQ_LIST, LOAD_FAQ_LIST_ERROR} = constants

describe(`handle ${LOAD_FAQ_LIST} action`, () => {
  it('returns the same state on an unhandled action', () => {
    expect(handleLoadFaqList(initialState, {type: undefined})).toMatchSnapshot()
  })

  it(`handles ${LOAD_FAQ_LIST} success action`, () => {
    expect(handleLoadFaqList(initialState, {type: LOAD_FAQ_LIST, payload: FaqData})).toEqual({
      ...initialState,
      faqLoaded: true,
      faqList: FaqData
    })

    expect(handleLoadFaqList(initialState, {type: LOAD_FAQ_LIST, payload: FaqData})).toMatchSnapshot()
  })
})


describe(`handle ${LOAD_FAQ_LIST_ERROR} action`, () => {
  it(`handles ${LOAD_FAQ_LIST_ERROR} action`, () => {
    const error = new Error('error')

    const errorMsg = [{'key': 0, 'text': 'Error: Faq Loading timeout'}];

    expect(handleLoadFaqList(initialState, {type: LOAD_FAQ_LIST, payload: errorMsg})).toEqual({
      ...initialState,
      faqLoaded: true,
      faqList: errorMsg
    })

    expect(handleLoadFaqList(initialState, {type: LOAD_FAQ_LIST_ERROR, error})).toMatchSnapshot()
  })
})