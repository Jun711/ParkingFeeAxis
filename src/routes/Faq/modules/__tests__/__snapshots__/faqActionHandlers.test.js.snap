// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle LOAD_FAQ_LIST action handles LOAD_FAQ_LIST success action 1`] = `
Object {
  "faqList": Array [
    Object {
      "key": "1",
      "text": "This app allows you to find nearby meter parking spot by searching for a location or move the center flag to a location.",
    },
    Object {
      "key": "2",
      "text": "The markers / flags on the map indicate a meter parking spot.",
    },
    Object {
      "key": "3",
      "text": "There are three types of parking flags. Green is the cheapest, yellow is the average while red is the most expensive nearby parking meter rate.",
    },
    Object {
      "key": "4",
      "text": "If you pan around the map, you will discover parking meter around a center point.",
    },
    Object {
      "key": "5",
      "text": "This app parking meter data comes from Vancouver Open Data Catalogue.",
    },
    Object {
      "key": "6",
      "text": "This app requires location permission and gps to work.",
    },
    Object {
      "key": "7",
      "text": "If no parking spots show up nearby a specific location, it may mean that there is no meter parkings nearby.",
    },
    Object {
      "key": "8",
      "text": "Your location data is not collected by this app.",
    },
    Object {
      "key": "9",
      "text": "If you find any bug or anomaly, feel free to contact me at ninja@ninja.com.",
    },
    Object {
      "key": "10",
      "text": "If you would like to see more features, feel free to contact me at ninja@ninja.com.",
    },
  ],
  "faqLoaded": true,
}
`;

exports[`handle LOAD_FAQ_LIST action returns the same state on an unhandled action 1`] = `
Object {
  "faqList": undefined,
  "faqLoaded": true,
}
`;

exports[`handle LOAD_FAQ_LIST_ERROR action handles LOAD_FAQ_LIST_ERROR action 1`] = `
Object {
  "faqList": undefined,
  "faqLoaded": true,
}
`;
